indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -145,11 +183,22 @@ export default function FormItem({
           disabled={disabled}
           isPurged={showPurged}
           htmlId={htmlId}
+          htmlName={htmlName}
           {...inputProps}
         />
       </>
     ) : null;
 
+  useEffect(() => {
+    if (!show && !setupMode) {
+      form.change(htmlName, undefined);
```

Comment:
The current logic here works nicely when creating a new registration, but not so much when editing a registration - by not sending anything, the field will not be cleared so they data remains.

This could either be handled client-side by setting the field's value to an empty value (which is currently not well-defined on the client side, it may be necessary to add it to registry.js), or (IMHO better) on the server side, ie clearing any field that's not active due to a condition (we have have an `empty_value` property on the field implementation class).

Reply:
OK, I managed to do it on the server side [f42cec1](https://github.com/indico/indico/pull/6678/commits/f42cec1a56bff15f95db7b38b194330a6d92cd2e).